openfasoc/generators/cryo-gen/tools/simulation.py

```diff
@@ -2,18 +2,26 @@
 import subprocess as sp
 
 
-def run_cryo_sim(simDir, lib_path, dut_path, sc_path) -> None:
+def run_cryo_sim(simDir, lib_path, dut_path, sc_path, prepex) -> None:
     # process 6-stage conv verilog
     with open(simDir + "templates/cryoInst_ngspice.sp", "r") as file:
         filedata = file.read()
         filedata = re.sub(r"@@PATH_TO_LIB", lib_path, filedata)
         filedata = re.sub(r"@@PATH_TO_DUT_SP", dut_path, filedata)
-        filedata = re.sub(r"@@PATH_TO_SC_SP", sc_path, filedata)
+        #filedata = re.sub(r"@@PATH_TO_SC_SP", sc_path, filedata)
+        if prepex:
+          filedata = re.sub(r"@@PATH_TO_RES", "cryoInst_prepex_res.ps", filedata)
+        else:
+          filedata = re.sub(r"@@PATH_TO_RES", "cryoInst_pex_res.ps", filedata)
+        
 
     with open(simDir + "/cryoInst_ngspice.sp", "w") as file:
         file.write(filedata)
 
-    print("Starting simulation")
+    if prepex:
+      print("Starting Prepex simulation")
+    else:
+      print("Starting Pex simulation")
 
     p = sp.Popen(
         [
```

Comment:
We can directly use ngspice instead of giving the full path. Also -r flag is used when have a raw file. Are we generating a raw file from simulation ?

Reply:
We don't need the raw file. I'll update this part.